flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java

```diff
@@ -110,6 +118,47 @@ public void testFailInEndOfConstructor() throws Exception {
                                 .build(EXECUTOR_RESOURCE.getExecutor()));
     }
 
+    @Test
+    public void streamTaskAsyncExceptionHandlerHandleExceptionInInitializingDuration()
+            throws Exception {
+        MockEnvironment mockEnvironment = MockEnvironment.builder().build();
+        RuntimeException expectedException = new RuntimeException("RUNTIME EXCEPTION");
+
+        mockEnvironment.setExpectedExternalFailureCause(AsynchronousException.class);
+        final String expectedErrorMessage = "EXPECTED_ERROR MESSAGE";
+
+        NoOpStreamTask noOpStreamTask =
+                new NoOpStreamTask(mockEnvironment) {
+                    volatile boolean handledAsyncException = false;
+
+                    @Override
+                    protected void init() throws Exception {
+                        super.init();
+
+                        CompletableFuture.runAsync(
+                                () -> {
+                                    this.handleAsyncException(
+                                            expectedErrorMessage, expectedException);
+                                    handledAsyncException = true;
+                                });
+                        while (!handledAsyncException) {
+                            ThreadUtils.sleep(Duration.ofMillis(10));
+                        }
+                    }
+                };
+        noOpStreamTask.invoke();
```

Comment:
How about call the `noOpStreamTask.restore()` due to this test checks whether AsyncException can be handled during restoring?

Please correct me, if `noOpStreamTask.restore()` doesn't work here.

Reply:
precise~